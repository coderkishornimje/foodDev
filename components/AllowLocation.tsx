import {Text, StyleSheet, View, Image} from 'react-native';
import React, {Component} from 'react';
import {TouchableOpacity} from 'react-native-gesture-handler';

interface IState {}
interface IProps {
  navigation?: any;
}
export default class AllowLocation extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.child1}>
          <Image
            style={{height: 300, width: 350}}
            source={require('../images/locationLogo.png')}
            resizeMode="stretch"
          />
          <Text style={styles.heading}>Allow Location</Text>
          <Text style={styles.subheading}>
            We need your Permeission to access
          </Text>
          <Text style={styles.subheading}>your location</Text>
        </View>
        <TouchableOpacity testID='allowBtn'
          style={styles.loginbtn}
          onPress={() => this.props.navigation.navigate('Login')}>
          <Text style={styles.logintext}>ALLOW LOCATION</Text>
        </TouchableOpacity>
        <Text
          style={{
            fontSize: 20,
            fontWeight: 'bold',
            color: '#000000',
            alignSelf: 'center',
            bottom: 15,
          }}>
          Don't Allow
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: '100%',
    width: '100%',
    backgroundColor: '#ffffff',
  },
  child1: {
    justifyContent: 'center',
    alignItems: 'center',
    margin: 50,
  },
  heading: {
    fontSize: 30,
    fontWeight: 'bold',
    color: '#000000',
  },
  subheading: {
    fontSize: 18,
    color: '#000000',
  },
  logintext: {
    fontSize: 20,
    alignSelf: 'center',
    //fontWeight:'bold',
    color: '#ffffff',
  },
  loginbtn: {
    backgroundColor: '#94CD00',
    marginHorizontal: 40,
    // left: -40,
    top: 0,
    padding: 15,
    borderRadius: 40,
    margin: 40,
  },
});
