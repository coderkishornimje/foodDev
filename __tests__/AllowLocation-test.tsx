import {
    fireEvent,
    render,

  } from '@testing-library/react-native';
  import renderer from 'react-test-renderer';
import AllowLocation from '../components/AllowLocation';

 

describe('home component', () => {
    it('renders correctly', () => {
      renderer.create(<AllowLocation />);
    });
    it('to check app render correctly', () => {
      const appRender = render(<AllowLocation />);
      expect(appRender).toMatchSnapshot();
      expect(appRender).toBeDefined();  
    });

    it('check permission button',()=>{
      const navigation={navigate:jest.fn()}
      const {getByTestId}=render(<AllowLocation navigation={navigation}/>);
        const allow= getByTestId('allowBtn');
        fireEvent.press(allow);
        expect(navigation.navigate).toHaveBeenCalledTimes(1);
    });

})  