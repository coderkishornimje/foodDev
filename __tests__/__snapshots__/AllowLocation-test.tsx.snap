// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component to check app render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "margin": 50,
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={
        {
          "testUri": "../../../images/locationLogo.png",
        }
      }
      style={
        {
          "height": 300,
          "width": 350,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Allow Location
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
        }
      }
    >
      We need your Permeission to access
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
        }
      }
    >
      your location
    </Text>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="allowBtn"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#94CD00",
          "borderRadius": 40,
          "margin": 40,
          "marginHorizontal": 40,
          "opacity": 1,
          "padding": 15,
          "top": 0,
        }
      }
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
          }
        }
      >
        ALLOW LOCATION
      </Text>
    </View>
  </RNGestureHandlerButton>
  <Text
    style={
      {
        "alignSelf": "center",
        "bottom": 15,
        "color": "#000000",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Don't Allow
  </Text>
</View>
`;
